gr-fec/lib/async_decoder_impl.h

```diff
@@ -34,11 +34,10 @@ class FEC_API async_decoder_impl : public async_decoder
 
     size_t d_max_bits_in;
     volk::vector<float> d_tmp_f32;
-    volk::vector<int8_t> d_tmp_u8;
+    volk::vector<uint8_t> d_tmp_u8;
     volk::vector<uint8_t> d_bits_out;
 
-    void decode_packed(pmt::pmt_t msg);
-    void decode_unpacked(pmt::pmt_t msg);
+    void decode(const pmt::pmt_t& msg);
```

Comment:
Following the discussion here, a `pmt::pmt_t` is a `shared_ptr`.
I suggest to change the signature to `void decode(const pmt::pmt_t msg);`. This change would make the signature a bit more obvious.

Reply:
This is somewhat above my knowledge of copy vs. move in C++, so asking to make sure: if the signature was `void decode(const pmt::pmt_t msg)` instead of `void decode(const pmt::pmt_t& msg)`, wouldn't that cause a copy of the `pmt_t` object? @marcusmueller was concerned that a `void decode(pmt::pmt_t msg)` would cause a needless copy (i.e., increase of the refcount of the `shared_ptr`). Does adding the `const` qualifier already elide the copy?

Reply:
I just [checked the documentation `typedef std::shared_ptr<pmt_base> pmt::pmt_t`](https://www.gnuradio.org/doc/doxygen/namespacepmt.html#a94a208b58dc16675b02674c28784d88b).

It does require some refcount accounting. However, [SO discussions on the subject](https://stackoverflow.com/a/8385581) basically point to our example and tell us: "Use the `shared_ptr` not the reference".

Reply:
The comments on that stackoverflow to me say:
* Pass a `shared_ptr` by value, then a copy is made. Pro: the copy keeps the allocation alive. Con: the copy is slightly costly because it involves increasing the refcounter atomically. They don't say what happens if you pass by `const` value, so I guess it's the same?
* Pass a `shared_ptr` as a `const` reference, then a copy is not made. Pro: it is cheap. Con: if whoever owns the `shared_ptr` deallocates the object, then your reference is dangling.

In this case I think that the caller to `decode()` owns the `shared_ptr`, and will keep it alive while `decode()` runs. Therefore, it seems harmless to pass as a `const` reference, and it is slightly cheaper. I think this is what @marcusmueller had in mind (please confirm).